presto-main/src/main/java/com/facebook/presto/operator/scalar/MapConcatFunction.java

```diff
@@ -139,42 +142,36 @@ public static Block mapConcat(MapType mapType, Object state, Block[] maps)
             return maps[lastMapIndex];
         }
 
+        Type keyType = mapType.getKeyType();
+        Type valueType = mapType.getValueType();
+        OptimizedTypedSet typedSet = new OptimizedTypedSet(keyType, maps.length, entries / 2, FUNCTION_NAME);
```

Comment:
`entries / 2` is very confusing to me. Maybe when compute `entries` in line 135, just /2 there?

Reply:
Agree it's confusing. This is because `SingleMapBlock` counts the positionCount for both key and value block. But line 135 is in the loop and I tend to not add more computation there. I added the following comments
```
// We need to divide the entries by 2 because the maps array is SingleMapBlocks and it had the positionCount twice as large as a normal Block
OptimizedTypedSet typedSet = new OptimizedTypedSet(keyType, maps.length, entries / 2, FUNCTION_NAME);
```